src/components/App/StepComponentMap.js

```diff
@@ -1,36 +1,53 @@
 import { h, Component } from 'preact'
 import Welcome from '../Welcome'
 import Select from '../Select'
-import {DocumentCapture, FaceCapture} from '../Capture'
+import {FrontDocumentCapture, BackDocumentCapture, FaceCapture} from '../Capture'
 import Complete from '../Complete'
 
-const stepToComponents = (stepDefaultOptions, {type: stepType, options: stepOptions}) => {
-  const optionExt = {...stepOptions, ...stepDefaultOptions};
-  switch (stepType) {
-    case 'document':
-      return [<Select method='document' {...optionExt} />,
-              <DocumentCapture {...optionExt} />]
-    case 'face':
-      return <FaceCapture {...optionExt}/>
-    case 'welcome':
-      return <Welcome {...optionExt} />
-    case 'complete':
-      return <Complete {...optionExt} />
-    default:
-      return <div>Step "{stepType}" does not exist</div>
+export const steps = (stepOptions, documentType) => {
+  const mapOptions = stepOption => stepOptionToStep(stepOption, documentType)
+  return shallowFlatten(stepOptions.map(mapOptions))
+}
+
+export const components = (stepList, componentOptions) => {
+  const mapSteps = step => stepToComponents(step, componentOptions)
+  return stepList.map(mapSteps)
+}
+
+const stepOptionToStep = (stepOption, documentType) => {
+  const step = formatStep(stepOption)
+  return step.type === 'document' ? documentStepsList(step.options, documentType) : step
+}
+
+const documentStepsList = (options, documentType) => {
+  const steps = [{type: 'documentSelect', ...options},
+                 {type: 'documentFront', ...options}]
+  if (['driving_licence', 'national_identity_card'].includes(documentType)) {
+    steps.push({type: 'documentBack', ...options})
   }
-};
-
-const typeToStep = type => {return {type}};//{type} will not return an object with the property type, because {} are also used to establish a multi line function
-
-const isStep = val => typeof val === 'object';
-
-const formatStep = typeOrStep => isStep(typeOrStep) ?  typeOrStep : typeToStep(typeOrStep);
+  return steps
+}
+
+const stepToComponents = (step, componentOptions) => {
+  const options = {...step, ...componentOptions}
+  const stepMap = {
+    welcome: <Welcome {...options}/>,
+    face: <FaceCapture {...options}/>,
+    documentSelect: <Select {...options}/>,
+    documentFront: <FrontDocumentCapture {...options}/>,
+    documentBack: <BackDocumentCapture {...options}/>,
+    complete: <Complete {...options}/>
+  }
```

Comment:
Best to do something more like `complete: () => <Complete {...options} />`, and then `return stepMap[step.type]()`

Just to avoid creating all of the possible steps each time this method is called, but only actually ever using one!